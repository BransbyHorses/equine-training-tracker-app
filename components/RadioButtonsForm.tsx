import {
	FormControl,
	FormControlLabel,
	Radio,
	RadioGroup,
	} from "@mui/material";
import React, { useEffect } from "react";

export default function RadioButtonsForm (props: any) {

    return (
        <FormControl>
        <RadioGroup
            defaultValue="radioform"
            name="radio-buttons-group"
            onChange={props.handleChange}>
            {props.items.map(({id, name}:any) => {
                return (
            <FormControlLabel 
                key={id} 
                value={name} 
                control={<Radio/>} 
                label={name}
                 />
                )
            }
            )
        }
        </RadioGroup>
    </FormControl>
    )
}

    