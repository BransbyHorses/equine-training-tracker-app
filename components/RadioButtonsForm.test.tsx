import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import RadioButtonsForm from "./RadioButtonsForm";

describe("RadioButtonsForm", () => {
	it("renders with minimum props", () => {
		const props = {
			handleChange: () => null,
			items: [],
		};
		const tree = renderer.create(<RadioButtonsForm {...props} />);
		expect(tree).toMatchSnapshot();
	});
});
