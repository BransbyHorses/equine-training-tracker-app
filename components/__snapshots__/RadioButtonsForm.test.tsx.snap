// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonsForm renders with minimum props 1`] = `
<div
  className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <div
    className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    role="radiogroup"
  />
</div>
`;
